<p>The databases that are available for searching are organized into several groups for ease of use:</p>
<table>
	<thead>
		<tr>
			<th>Database name</th>
			<th>Database type</th>
			<th>Description</th>
		</tr>
	</thead>
	<tbody>
		<tr>
			<td><em>Lotus japonicus</em> Gifu mRNA</td>
			<td>Nucleotide</td>
			<td>De novo assembled transcript contigs from Illumina non-directional mRNA-seq data. 58,527 sequences; 40,650,058 total letters.</td>
		</tr>
		<tr>
			<td><em>Lotus japonicus</em> MG20 Illumina- and Sanger-based assembly</td>
			<td>Nucleotide</td>
			<td>The program PCAP.rep was used to assemble Illumina contigs (LotusK31ver2GC_illumina_contigs) with the Sanger shotgun data (LjSGA).</td>
		</tr>
		<tr>
			<td><em>Lotus japonicus</em> MG20 PacBio contigs</td>
			<td>Nucleotide</td>
			<td>.ctg are the high confidence contigs. PacBio is a sequencing technology that generates long (up to 10 kb) but error-prone (up to 20%) reads. The PacBio reads have been corrected using Illumina data prior to contig assembly.</td>
		</tr>
		<tr>
			<td><em>Lotus japonicus</em> MG20 PacBio unitigs</td>
			<td>Nucleotide</td>
			<td>.utg are all the assembled PacBio contigs (unitigs). PacBio is a sequencing technology that generates long (up to 10 kb) but error-prone (up to 20%) reads. The PacBio reads have been corrected using Illumina data prior to contig assembly.</td>
		</tr>
		<tr>
			<td><em>Lotus japonicus</em> MG20 mRNA</td>
			<td>Nucleotide</td>
			<td>De novo assembled transcript contigs from Illumina non-directional mRNA-seq data. 57,916 sequences; 39,463,161 total letters.</td>
		</tr>
		<tr>
			<td><em>Lotus japonicus</em> MG20 raw Illumina-based assembly</td>
			<td>Nucleotide</td>
			<td>Genomic MG20 contigs assembled using SOAP de novo based on four Illumina libraries with different insert sizes. Raw and uncurated assembly. Contigs may overlap. 904,251 sequences; 439,303,697 total letters.</td>
		</tr>
		<tr>
			<td><em>Lotus japonicus</em> MG20 v3.0 CDS</td>
			<td>Nucleotide</td>
			<td>CDS library of Lotus japonicus MG20. 50,599 sequences; 45,983,720 total letters.</td>
		</tr>
		<tr>
			<td><em>Lotus japonicus</em> MG20 v3.0 cDNA</td>
			<td>Nucleotide</td>
			<td>cDNA library of Lotus japonicus MG20. 98,043 sequences; 108,691,465 total letters.</td>
		</tr>
		<tr>
			<td><em>Lotus japonicus</em> MG20 genome v2.4</td>
			<td>Nucleotide</td>
			<td><em>Lotus</em> genome release 2.4. This is the genome version that matches the Apollo browser GFF files in the folder "-Apollo favourite genes/2010_09_06_GFF" on the commom server. According to Shusei, there should only be minor changes to the coordinates on a part of chr5 between version 2.4 and 2.5. There are large differences with respect to the annotation between the two versions due to removal of a lot of retro-element-related sequence annotation in v2.5.</td>
		</tr>
		<tr>
			<td><em>Lotus japonicus</em> MG20 genome v2.5</td>
			<td>Nucleotide</td>
			<td><em>Lotus</em> genome release 2.5, only sequences anchored on chromosomes, including chr0 long contigs with unknown position and the chloroplast sequence. 8 sequences; 301,492,348 total letters.</td>
		</tr>
		<tr>
			<td><em>Lotus japonicus</em> MG20 genome v2.5 + LjSGA</td>
			<td>Nucleotide</td>
			<td>As lj_r25 but including also all the short sanger shotgun contigs (LjSGAs), which are not anchored on chromosomes. 67,748 sequences; 397,315,927 total letters.</td>
		</tr>
		<tr>
			<td><em>Lotus japonicus</em> MG20 genome v2.8</td>
			<td>Nucleotide</td>
			<td><em>Lotus</em> genome pre-release 2.8. This is not an official release but the current state of the new <em>Lotus</em> genome assembly. Overlaps between contigs still need to be checked, and the coordinates will not match the version to be published. Contains only sequences anchored on the six <em>Lotus</em> chromosomes. 6 sequences; 295,202,327 total letters.</td>
		</tr>
		<tr>
			<td><em>Lotus japonicus</em> MG20 genome v3.0</td>
			<td>Nucleotide</td>
			<td>Version 3.0 of the <em>Lotus</em> genome including chr0 with unanchored contigs, chr1&dash;6, chloroplast and mitochondrion sequences. 9 sequences; 447,516,816 total letters.</td>
		</tr>
		<tr>
			<td><em>Lotus japonicus</em> probes</td>
			<td>Nucleotide</td>
			<td><p>The probes used to generate the <a href="http://ljgea.noble.org/v2/">LjGEA dataset</a>. The naming nomenclature for these probes&dash;when ending with&hellip;</p><ul>
				<li><code>_at</code>: Anti-sense target<li>
				<li><code>_st</code>: Sense target</li>
				<li><code>_s_at</code>: Probeset shares common probes among multiple transcripts (originated from multiple genes)</li>
				<li><code>_a_at</code>: Gene may code for alternatively-spliced transcripts</li>
				<li><code>_x_at</code>: It was not possible to design unique probes for multiple transcripts (this probeset is not unique for the gene)</li>
			</ul></td>
		</tr>
		<tr>
			<td><em>Ramularia collo-cygni</em> DK05 contigs</td>
			<td>Nucleotide</td>
			<td>Contig library of <em>Ramularia collo-cygni</em> from batch DK05.</td>
		</tr>
		<tr>
			<td>DeepSAGE tag sequences</td>
			<td>Nucleotide</td>
			<td>Transcript contigs from <em>L. japonicus</em> Gene index 6.0. These TCs were used for annotation of tags from the DeepSAGE experiment. 43,180 sequences; 30,997,719 total letters.</td>
		</tr>
		<tr>
			<td>KAW12 scaffolds</td>
			<td>Nucleotide</td>
			<td>Genome sequence of the KAW endophyte worked on by Simona and Rafal. 7 sequences; 7,876,853 total letters.</td>
		</tr>
		<tr>
			<td>KAW12 translations</td>
			<td>Nucleotide</td>
			<td>Amino acid sequence of the endophyte <em>Rhizobium</em> KAW12 genome. 7574 sequences; 2,292,824 total letters.</td>
		</tr>
		<tr>
			<td><em>Lotus japonicus</em> MG20 v3.0 Proteins</td>
			<td>Protein</td>
			<td>Protein library of <em>Lotus japonicus</em> MG20. 50,599 sequences; 15,289,260 total letters.</td>
		</tr>
	</tbody>
</table>